### DISCUSSION TITLE
Identityserver per tenant options loginurl

### DISCUSSION THREAD

**Discussion by fbjerggaard** [Finbuckle/Finbuckle.MultiTenant#592] [Q&A] [answered]
Hi,

I am trying to create a multi-tenant identityserver setup with a different login URL per tenant.

I am using a SPA for the login-part of the process, and as such need to redirect to `<tenant>.auth.example.net` so that my SPA knows which tenant to login to.

I have tried the perTenantOptions for both `UserInteractionOptions` and `IdentityServerOptions` but to no avail:
```
.WithPerTenantOptions<UserInteractionOptions>((options, t) =>
{
    options.LoginUrl = $"{builder.Configuration["AuthSettings:FrontendUrl"].Replace("{__tenant__}", t.Identifier)}/auth/login";
    options.LogoutUrl = $"{builder.Configuration["AuthSettings:FrontendUrl"].Replace("{__tenant__}", t.Identifier)}/auth/logout";
    options.ErrorUrl = $"{builder.Configuration["AuthSettings:FrontendUrl"].Replace("{__tenant__}", t.Identifier)}/auth/error";
})
```

Identityserver will redirect to the local `/Auth/Login` and not use the per-tenant options.

It seems like the options are being cached by idsrv and I don't see any way of overriding these after identityserver has been started.

Am I missing something, or is what I am trying to do not possible?

**Comment by AndrewTriesToCode** [comment, marked as answer]
Hi, unfortunately you are not missing anything. I've dug into it and they aren't currently using the Options pattern the way ASP.NET presents it, and they do cache things. 

Your best bet is to look at their end point implementations and make your own logic customizations there.

**Reply by fbjerggaard** [reply]
Thanks for confirming it's not me who's stupid :) I found another way to implement what I wanted by utilizing acr_values instead.

And thanks for a great library!

**Reply by AndrewTriesToCode** [reply]
acr_values is another good option for anyone else who stumbles across this conversation in the future
